### PR TITLE
Ignore uri_has_not_been_generated errors.

### LINKED ISSUE #213: Files that import .template.dart files that are generated by Angular produce failures in Strong Mode analysis.
Example:
https://pub.dartlang.org/packages/angular_components/versions/0.9.0-alpha%2B2#-analysis-tab-

The `uri_has_not_been_generated: ignore` option I have in the `analysis_options.yaml` file is not being used. https://github.com/dart-lang/angular_components/blob/master/analysis_options.yaml#L8

### PR DESCRIPTION
**Pull request by isoos**
Fixes #213

### CODE REVIEW

**Comment by isoos** [comment]
I still think that this is a bad precedent.

File: lib/src/analysis_options.dart [hunk 1]
```diff
@@ -5,6 +5,8 @@
 const String analysisOptions = '''
 analyzer:
   strong-mode: true
+  errors:
+    uri_has_not_been_generated: ignore
```
Comment (kevmoo) [inline lib/src/analysis_options.dart]: No!

I think the idea is: if the source package has this setup – then copy it.

Just whitelist `uri_has_not_been_generated` for pass-through

File: lib/src/analysis_options.dart [hunk 2]
```diff
@@ -48,3 +52,35 @@ linter:
     - unrelated_type_equality_checks
     - valid_regexps
 ''';
+
+const _analyzerErrorKeys = const <String>['uri_has_not_been_generated'];
+
+String customizeAnalysisOptions(String original, bool isFlutter) {
```
Comment (kevmoo) [inline lib/src/analysis_options.dart]: TODO: add an unit test for this – since it's more than a bit complicated

Comment (kevmoo) [reply lib/src/analysis_options.dart]: @isoos – did you sanity check it w/ one of the samples we care about?

Comment (isoos) [reply lib/src/analysis_options.dart]: Added tests, which caught an edge-case that was not caught with the sampling manual test.

Comment (kevmoo) [reply lib/src/analysis_options.dart]: 😄
